lightningd/peer_control.c

```diff
@@ -1343,6 +1343,89 @@ static const struct json_command getinfo_command = {
 };
 AUTODATA(json_command, &getinfo_command);
 
+
+static struct command_result *json_setchannelfee(struct command *cmd,
+					 const char *buffer,
+					 const jsmntok_t *obj UNNEEDED,
+					 const jsmntok_t *params)
+{
+	struct json_stream *response;
+	const jsmntok_t *idtok;
+	unsigned int *base;
+	unsigned int *ppm;
+	struct peer *peer;
+	/* FIXME: gcc 7.3.0 thinks this might not be initialized. */
+	struct channel *channel = NULL;
+	struct channel_id cid;
+	struct command_result *res;
+
+	if (!param(cmd, buffer, params,
+		   p_req("id", param_tok, &idtok),
+		   p_opt_def("base", param_number, &base, 0),
+		   p_opt_def("ppm", param_number, &ppm, 0),
+		   NULL))
+		return command_param_failed();
+
+	/* find the channel */
+	peer = peer_from_json(cmd->ld, buffer, idtok);
+	if (peer) {
+		channel = peer_active_channel(peer);
+		if (!channel)
+			return command_fail(cmd, LIGHTNINGD,
+				    "Could not find active channel of peer with that id");
+	} else {
+		res = command_find_channel(cmd, buffer, idtok, &channel);
+		if (res)
+			return res;
+	}
+
+	/* check channel is found and in valid state */
+	if (!channel)
+		return command_fail(cmd, LIGHTNINGD, "Could not find peer with that id");
```

Comment:
Move this check up before '}' and change message to "Could not find channel with that id"?

Reply:
okay did so, but now its actually a "cannot happen exception", since `comman_find_channel` will always have channel set if no error was returned.